TheAmazingAudioEngine/Utilities/AEIOAudioUnit.m

```diff
@@ -48,6 +49,69 @@ @interface AEIOAudioUnit ()
 #endif
 @end
 
+
+////////////////////////////////////////////////////////////////////////////////
+// sets nr of seconds between reports, 0 = no reporting
+#define AE_REPORT_TIME 2
+
+#if AE_REPORT_TIME
+
+static OSStatus notifyCallback(
+	void *							inRefCon,
+	AudioUnitRenderActionFlags *	ioActionFlags,
+	const AudioTimeStamp *			inTimeStamp,
+	UInt32							inBusNumber,
+	UInt32							inNumberFrames,
+	AudioBufferList * __nullable	ioData)
+{
+	static UInt64 startTime = 0;
+	static UInt64 finishTime = 0;
+	
+	if (*ioActionFlags & kAudioUnitRenderAction_PreRender)
+	{
+		startTime = mach_absolute_time();
+	}
+	else
+	if (*ioActionFlags & kAudioUnitRenderAction_PostRender)
+	{
+		finishTime = mach_absolute_time();
+		
+		double renderTime = AESecondsFromHostTicks(finishTime - startTime);
+		
+		// Compute short term average time
+		static double avgTime = 0;
+		avgTime += 0.1 * (renderTime - avgTime);
+		
+		// Compute maximum time since last report
+		static double maxTime = 0;
+		if (maxTime < renderTime)
+		{ maxTime = renderTime; }
+		
+		// Check report frequency
+		static double lastTime = 0;
+		double time = AECurrentTimeInSeconds();
+		if (lastTime + AE_REPORT_TIME <= time)
+		{
+			lastTime = time;
+			double reportTime1 = avgTime;
+			double reportTime2 = maxTime;
+			dispatch_async(dispatch_get_main_queue(), \
+			^{
+				NSLog(@"Render time avg = %lfs)", reportTime1);
+				NSLog(@"Render time max = %lfs)", reportTime2);
+			});
```

Comment:
As far as I know, the render notify callback runs on the audio thread. Unfortunately dispatch_async is not real-time safe and should not be used here. But I see that the idea here is for debug purposes only, perhaps AE_REPORT_TIME should only be set if DEBUG? And then you don't need dispatch_async either, NSLog can be used from the audio thread too (for debugging, not in production code).